hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -327,8 +327,10 @@ protected void completeCompaction(HoodieCommitMetadata metadata, HoodieTable tab
     try {
       this.txnManager.beginTransaction(Option.of(compactionInstant), Option.empty());
       finalizeWrite(table, compactionCommitTime, writeStats);
-      // commit to data table after committing to metadata table.
-      writeTableMetadata(table, compactionCommitTime, metadata, context.emptyHoodieData());
+      // if metatable is enable, then commit to data table after committing to metadata table.
+      if (config.getMetadataConfig().enabled()) {
+        writeTableMetadata(table, compactionCommitTime, metadata, context.emptyHoodieData());
+      }
       LOG.info("Committing Compaction " + compactionCommitTime + ". Finished with result " + metadata);
```

Comment:
`writeTableMetadata` already has a inline check for the disability of the MDT.

Reply:
writeTableMetadata judge whether to execute commit data to metadata table had two conditions:
**1. mdt enable  2. mdt dir exists;**
user if want to stop commit to metadata who enable it before(in the condition metadata dir would exists)，can not stop compaction operation commit data to metadata by set hoodie.metadata.enable=false，currently could not support it which is not fitable @danny0405 @CTTY

Reply:
Maybe add a new config to control it could be better

Reply:
The master code already fixes this: https://github.com/apache/hudi/pull/9546

Reply:
OK close the pr